src/odemis/model/_components.py

```diff
@@ -643,29 +653,37 @@ def __init__(self, name, role, **kwargs):
     def updateMetadata(self, md):
         Detector.updateMetadata(self, md)
         mdf = self._metadata
-        if self.pixelSize is not None:
-            try:
-                pxs = self.pixelSize.value[0]  # pixel should be square
-                mag = mdf[_metadata.MD_LENS_MAG]
-                na = mdf[_metadata.MD_LENS_NA]
-                ri = mdf[_metadata.MD_LENS_RI]
-                l = 550e-9  # the light wavelength
-                # We could use emission wavelength center for l, but it's mostly
-                # confusing for the user that the focus sensitivity changes when
-                # the observed part changes. So just use 550 nm, which is never
-                # more than 50% wrong.
-                # from https://www.microscopyu.com/articles/formulas/formulasfielddepth.html
-                dof = (l * ri) / na ** 2 + (ri * pxs) / (mag - na)
-                rng = self.depthOfField.range
-                if rng[0] <= dof <= rng[1]:
-                    self.depthOfField._set_value(dof, force_write=True)
-                else:
-                    logging.warning("Depth of field computed seems incorrect: %f m", dof)
-            except KeyError:
-                # Not enough metadata is present for computing Depth of Field
-                pass
-            except Exception:
-                logging.warning("Failure to update the depth of field", exc_info=True)
+
+        try:
+            pxs = self.pixelSize.value[0]  # pixel should be square
+            mag = mdf[_metadata.MD_LENS_MAG]
+            na = mdf[_metadata.MD_LENS_NA]
+            ri = mdf[_metadata.MD_LENS_RI]
+            l = 550e-9  # the light wavelength
+            # We could use emission wavelength center for l, but it's mostly
+            # confusing for the user that the focus sensitivity changes when
+            # the observed part changes. So just use 550 nm, which is never
+            # more than 50% wrong.
+        except (AttributeError, KeyError):
+            # Not enough metadata is present for computing Depth of Field and spot size.
+            return
+
+        try:
+            # from https://www.microscopyu.com/articles/formulas/formulasfielddepth.html
+            dof = (l * ri) / na ** 2 + (ri * pxs) / (mag - na)
+            self.depthOfField._set_value(dof, force_write=True)
+        except TypeError:
+            logging.warning("Depth of field computed seems incorrect: %f m", dof)
+        except Exception:
+            logging.warning("Failure to update the depth of field", exc_info=True)
+
+        try:
+            sigma = (mag / pxs) * synthetic.psf_sigma_wffm(ri, na, l)
+            self.spotSize._set_value(sigma, force_write=True)
+        except TypeError:
+            logging.warning("Spot size computed seems incorrect: %f px", sigma)
```

Comment:
That warning is a little tricky because if the first line (in the try block) triggers a `TypeError` (which is not so unlikely as it can happen that values are passed as string instead of float, then sigma is undefined, which would trigger an unhandled exception. Moreover, if the value is out of range, a `IndexError` is raised, and I think it should also be handle here.

I think the easiest is to place `_set_value()` in a second try block, with just this `except`.

(same comment for the new depthOfField code.

Reply:
That first line can only trigger a `TypeError` if the entries in `self._metadata` have the wrong type. That should not happen, right? Or phrased differently, would an unhandled exception in that case not be the desired behaviour?

Reply:
The metadata can be written by the user (via updateMetadata() or `odemis-cli --update-metadata`, or the microscope file). So it can be the wrong type. In such case, it's preferable to trigger a TypeError, not a `NameError("name sigma is undefined")` on the `logging.warning(...)` line, as that would be hard to understand for the user.
Note that for the specific metadata here, typically the user is supposed to update them via the VA on the lens, which is then copied into a metadata. So it's not so likely to happen, but still better to report properly.